examples/react-hooks-example/src/App.js

```diff
@@ -0,0 +1,134 @@
+import React, {useEffect, useState} from 'react';
+import {
+    Box,
+    Grommet,
+    DataTable,
+    Text
+} from 'grommet';
+import { pipe } from 'rxjs';
+import Subspace, {$latest} from '@embarklabs/subspace';
+import Web3 from 'web3';
+import exchangeABI from './contract/exchange_abi.json'
+
+const web3 = new Web3('https://cloudflare-eth.com');
+const subspace = new Subspace(web3.currentProvider);
+var dai = new web3.eth.Contract(exchangeABI, '0x2a1530C4C41db0B0b2bB646CB5Eb1A67b7158667');
+const daiContract = subspace.contract(dai);
+subspace.init();
+
+const theme = {
+    global: {
+        font: {
+            family: 'Roboto',
+            size: '18px',
+            height: '20px'
+        }
+    }
+};
+
+
+function App() {
+
+    const [txnObserver, setObservable] = useState();
+    const [last5Observer, setLast5Observer] = useState();
+    const [latestBlock, setBlock] = useState();
+    const [last5, setLast5] = useState([]);
+
+    //Trade details object for calculating exchange rate
+    function TradeDetails(tokensSold, ethBought) {
+        this.tokensSold = web3.utils.fromWei(tokensSold);
```

Comment:
Might be a good idea to have a guard in case `tokensSold` or `ethBought` is null/undefined/not a numeric string, because otherwise `web3.utils.fromWei` can throw an error.

Reply:
Will add appropriate try/catch blocks in next commit.